stock_picking_delivery_link/models/stock_picking.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Camptocamp SA (http://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+def get_first_move_dest(moves, done=False):
+    move_states = ("cancel", "done")
+    for move in moves.move_dest_ids:
+        if move.state in move_states if done else move.state not in move_states:
```

Comment:
as discussed in "private", you could use `if not operator.xor(done, move.state in move_states)` instead of the double if.

Reply:
can't you simply change moves_states?
```py
move_states = ("cancel", "done")
if not done:
    move_states = (a, b, c)
if move.state in move_states:
```
sounds more clear and explicit

Reply:
As seen together, better to go with an "if box", much more readable.

Reply:
@simahawk I didn't want to list all possible states of a move, I think the code is OK now.